### PR TITLE
Make URI cells clickable

### LINKED ISSUE #1322: Make URL cells clickable
## Reproduce

1. "New Table" > "Import Data" > "Copy and Paste Text" > paste the following

    ```text
    project,repo
    Mathesar,https://github.com/centerofci/mathesar
    ```

1. Load the table

1. Expect to find a way to click on the URL in the cell and load the page in my browser

1. Observe the URL is not clickable

## Specs

- The overall behavior should be similar to AirTable, as in this example:

    Peek 2022-04-27 12-33

- The URL should only be clickable when the cell is selected

- Only the text itself should be clickable -- not the padding around it. I should still be able to click in the cell but below the text. This way I can enter edit mode via double click, even with the cell is selected (if I double click within the padding).

- The URL should be underlined while viewing, but not while editing.

- The text color should be black when the link is not clickable and blue when the link is clickable.

- Clicking on the link should open the page in a new tab

### PR DESCRIPTION
**Pull request by seancolsen**
Fixes #1322

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: mathesar_ui/src/stores/abstract-types/type-configs/uri.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    type: 'uri',
+  },
+  getDbConfig: () => ({
+    form: dbForm,
+    determineDbTypeAndOptions: () => ({
+      dbType: DB_TYPES.MATHESAR_TYPES__URI,
+      typeOptions: null,
+    }),
+    constructDbFormValuesFromTypeOptions: () => ({}),
+  }),
```
Comment (pavish) [inline mathesar_ui/src/stores/abstract-types/type-configs/uri.ts:31]: This is optional. Not specifying `getDbConfig` will work as expected.

**Review by pavish — CHANGES_REQUESTED** [review]
@seancolsen The PR does not seem to cover all the cases mentioned in the issue.

* The URL should only be clickable when the cell is selected
* The text color should be black when the link is not clickable and blue when the link is clickable.

I think the first case is important to handle, since trying to click the cell to activate it, will almost always open a new tab, which would be quite annoying.

File: mathesar_ui/src/stores/abstract-types/type-configs/uri.ts (hunk 1 above)
Comment (seancolsen) [reply mathesar_ui/src/stores/abstract-types/type-configs/uri.ts]: Fixed in dd832d8e7a5bff93f7579328de75bdc6ceada701

**Review by pavish — APPROVED** [review]
@seancolsen Looks good.

There is one small bug. The content is underlined for all values including NULL.

Screenshot 2022-05-12 at 6 26 38 PM

Feel free to fix this in this PR or upcoming ones.

**Comment by seancolsen** [comment]
Ahh! To appease the CI, I had to come up with an icon for the URI type now. I did that in 824119b4342d2e2516cc81177d7876c48016e7eb. Praise TypeScript! This is the sort of situation where it really shines!

**Pull request merged**